src/us/kbase/common/test/controllers/mongo/MongoController.java

```diff
@@ -74,26 +74,33 @@ public Path getTempDir() {
 		return tempDir;
 	}
 	
-	public void destroy(boolean deleteTempFiles) throws IOException {
+	public void destroy(boolean deleteTempFiles) throws IOException, InterruptedException {
 		if (mongo != null) {
 			mongo.destroy();
 		}
 		if (tempDir != null && deleteTempFiles) {
-			FileUtils.deleteDirectory(tempDir.toFile());
+			try {
+				FileUtils.deleteDirectory(tempDir.toFile());
+			} catch (IOException e) {
+				// probably mongo deleted a file after the function listed it, race condition
+				Thread.sleep(1000);
+				// if it fails again just fail hard
+				FileUtils.deleteDirectory(tempDir.toFile());
+			}
 		}
 	}
 
 	public static void main(String[] args) throws Exception {
 		MongoController ac = new MongoController(
-				"/kb/runtime/bin/mongod",
+				"/home/crushingismybusiness/mongo/3.6.12/bin/mongod",
```

Comment:
presumably this needs to be changed for set ups that aren't on your laptop...

Reply:
`main` is just a helper for doing manual testing of this code and isn't used otherwise, so leaving it as is it ok - it could just be deleted with no negative repercussions other than that the next person who wants to manually test the controller would have to do something similar. Paths to the mongo executable are going to be different from machine to machine so now that /kb/runtime is not longer a thing there isn't a single path we can put here